### PR TITLE
Clean product form

### PR DESCRIPTION
**Pull request by mdi-odoo**
Task: https://www.odoo.com/web#id=28604&view_type=form&model=project.task&action=327&menu_id=4720

Pad: https://pad.odoo.com/p/r.5b67033ce668139d65a4d01ba3f7bdd0

### CODE REVIEW

File: addons/account/views/product_view.xml [hunk 1]
```diff
@@ -7,25 +7,28 @@
...
                                 <field name="property_account_expense_id"
                                     domain="[('internal_type','=','other'),('deprecated','=',False)]"
                                     groups="account.group_account_user"/>
-                                <field name="supplier_taxes_id" widget="many2many_tags"/>
                             </group>
                         </group>
                         <group name="accounting"/>
                     </page>
                 </page>
+                <xpath expr="//group[@name='email_template_and_project']" position="before">
```
Comment (mdi-odoo) [inline addons/account/views/product_view.xml]: You can directly write <group name="email_template_and_project" position="before"> .

File: addons/mrp/views/product_views.xml [hunk 2]
```diff
@@ -49,6 +49,8 @@
                         attrs="{'invisible':[('type', 'not in', ['product', 'consu'])]}" icon="fa-flask">
                         <field string="Bill of Materials" name="bom_count" widget="statinfo" />
                     </button>
+                    </button>
+                <button name="toggle_active" position="before">
```
Comment (mdi-odoo) [inline addons/mrp/views/product_views.xml]: Remove unnecessary changes.

File: addons/purchase/views/purchase_views.xml [hunk 3]
```diff
@@ -619,8 +632,8 @@
             <field name="inherit_id" ref="product.product_template_only_form_view"/>
             <field name="groups_id" eval="[(4, ref('purchase.group_purchase_user'))]"/>
             <field name="arch" type="xml">
-                <button name="toggle_active" position="before">
-                   <button class="oe_stat_button" name="%(purchase.action_purchase_line_product_tree)d"
+                  <button name="toggle_active" position="before">
```
Comment (mdi-odoo) [inline addons/purchase/views/purchase_views.xml]: Remove the unnecessary changes.

File: addons/product_expiry/models/product_product.py [hunk 4]
```diff
@@ -7,11 +7,11 @@
 class ProductTemplate(models.Model):
     _inherit = 'product.template'
 
-    life_time = fields.Integer(string='Product Life Time',
+    life_time = fields.Integer(string='Best within',
```
Comment (purnendu03) [inline addons/product_expiry/models/product_product.py]: We use camel case for labeling in odoo!
-Best within 
+Best Within

File: addons/account/models/product.py [hunk 5]
```diff
@@ -34,6 +34,12 @@ class ProductTemplate(models.Model):
         string="Expense Account", oldname="property_account_expense",
         domain=[('deprecated', '=', False)],
         help="This account will be used for invoices instead of the default one to value expenses for the current product.")
+    purchase_method = fields.Selection([
+        ('purchase', 'On ordered quantities'),
```
Comment (mdi-odoo) [inline addons/account/models/product.py]: Why **purchase_method** is moved from **purchase** to **account** module.

File: addons/account_asset/views/product_views.xml [hunk 6]
```diff
@@ -6,10 +6,11 @@
         <field name="model">product.template</field>
         <field name="inherit_id" ref="account.product_template_form_view"/>
         <field name="arch" type="xml">
-            <field name="property_account_expense_id" position="before">
+            <field name="property_account_expense_id" position="after">
                 <field name="asset_category_id"
                     domain="[('type', '=', 'purchase')]"
-                    context="{'default_type': 'purchase'}"/>
+                    context="{'default_type': 'purchase'}"
+                    groups="account.group_account_user,account.group_account_manager"/>
```
Comment (mdi-odoo) [inline addons/account_asset/views/product_views.xml]: May be **groups="account.group_account_user"** is enough .

File: addons/event_sale/data/event_sale_data.xml [hunk 7]
```diff
@@ -8,7 +8,6 @@
 
         <record id="product_product_event" model="product.product">
             <field name="list_price">10.0</field>
-            <field name="event_ok" eval="True"/>
             <field name="standard_price">30.0</field>
```
Comment (mdi-odoo) [inline addons/event_sale/data/event_sale_data.xml]: May be you forgot to change **type="event"**.

File: addons/purchase/views/purchase_views.xml [hunk 8]
```diff
@@ -586,7 +586,7 @@
             <field name="inherit_id" ref="account.view_category_property_form"/>
             <field name="arch" type="xml">
                 <field name="property_account_income_categ_id" position="before">
-                    <field name="property_account_creditor_price_difference_categ" domain="[('deprecated','=',False)]"/>
+                    <field name="property_account_creditor_price_difference_categ" domain="[('deprecated','=',False)]" groups="account.group_account_user,account.group_account_manager"/>
```
Comment (mdi-odoo) [inline addons/purchase/views/purchase_views.xml]: May be **groups="account.group_account_user"** is enough .

File: addons/purchase/views/purchase_views.xml [hunk 9]
```diff
@@ -599,13 +599,13 @@
...
-            <field name="name">product.template.account.purchase.ok.form.inherit</field>
+        <record id="view_product_purchase_ok_form" model="ir.ui.view">
+            <field name="name">product.template.purchase.ok.form.inherit</field>
             <field name="model">product.template</field>
-            <field name="inherit_id" ref="account.product_template_form_view"/>
+            <field name="inherit_id" ref="purchase.view_product_supplier_inherit"/>
             <field name="arch" type="xml">
                 <field name="property_account_expense_id" position="replace" >
-                     <field name="property_account_expense_id" domain="[('deprecated','=',False)]" attrs="{'readonly': [('purchase_ok', '=', 0)]}" groups="account.group_account_user"/>
+                     <field name="property_account_expense_id" domain="[('deprecated','=',False)]" attrs="{'readonly': [('purchase_ok', '=', 0)]}" groups="account.group_account_user,account.group_account_manager"/>
```
Comment (mdi-odoo) [inline addons/purchase/views/purchase_views.xml]: Same here.

File: addons/sale_stock/views/stock_views.xml [hunk 10]
```diff
@@ -34,5 +34,27 @@
             </field>
         </record>
 
+        <record id="product_template_form_view_sale_order_inherit_button" model="ir.ui.view">
+            <field name="name">product.template.inherit.form</field>
```
Comment (mdi-odoo) [inline addons/sale_stock/views/stock_views.xml]: Please follow the guidelines for naming conventions.

File: addons/sale_stock/views/stock_views.xml [hunk 11]
```diff
@@ -34,5 +34,27 @@
...
+            <field name="inherit_id" ref="sale.product_template_form_view_sale_order_button"/>
+            <field name="model">product.template</field>
+            <field name="arch" type="xml">
+                <xpath expr="//button[@name='action_view_sales'][last()]" position="replace" />
+            </field>
+        </record>
+
+        <record id="product_template_form_view_procurement_inherit_button" model="ir.ui.view">
```
Comment (mdi-odoo) [inline addons/sale_stock/views/stock_views.xml]: Same as above.

File: addons/procurement/views/product_views.xml [hunk 12]
```diff
@@ -10,8 +10,8 @@
                 <xpath expr="//button[@name='toggle_active']" position="before">
                    <button class="oe_stat_button" name="%(procurement.procurement_order_action_product_template_stat)d"
                            icon="fa-truck" string="Procurements" type="action" 
-                           groups="base.group_no_one"
-                           attrs="{'invisible':[('type', '=', 'service')]}"/>
+                           groups="stock.group_adv_location" 
+                           attrs="{'invisible':[('type', 'in', ['service', 'event'])]}"/>
```
Comment (mdi-odoo) [inline addons/procurement/views/product_views.xml]: Remove type = 'event', as its no more .

File: addons/account/views/product_view.xml [hunk 13]
```diff
@@ -7,19 +7,19 @@
             <field name="priority">5</field>
             <field name="inherit_id" ref="product.product_template_form_view"/>
             <field name="arch" type="xml">
-                <page name="sales" position="after">
-                    <page string="Invoicing" name="invoicing" groups="account.group_account_invoice">
+                <page name="inventory" position="after">
+                    <page string="Invoicing" name="invoicing">
                         <group name="properties">
-                            <group>
+                            <group string="Receivables" groups="account.group_account_user,account.group_account_manager">
```
Comment (mdi-odoo) [inline addons/account/views/product_view.xml]: Just **group="account.group_account_user"** is enough.

File: addons/account/views/product_view.xml [hunk 14]
```diff
@@ -7,19 +7,19 @@
...
                                 <field name="property_account_income_id"
-                                    domain="[('internal_type','=','other'),('deprecated','=',False)]"
-                                    groups="account.group_account_user"/>
-                                <field name="taxes_id" widget="many2many_tags"/>
+                                    domain="[('internal_type','=','other'),('deprecated','=',False)]"/>
                             </group>
-                            <group>
+                            <group string="Payables" groups="account.group_account_user,account.group_account_manager">
```
Comment (mdi-odoo) [inline addons/account/views/product_view.xml]: Same as above.

File: addons/sale/views/sale_views.xml [hunk 15]
```diff
@@ -609,6 +609,25 @@
...
                 </button>
+                <group name="warning" position="inside">
+                    <group string="Warning when Selling this Product" groups="sale.group_warning_sale">
+                        <field name="sale_line_warn" nolabel="1"/>
+                        <field name="sale_line_warn_msg" colspan="3" nolabel="1"
+                                attrs="{'required':[('sale_line_warn','!=','no-message')],'readonly':[('sale_line_warn','=','no-message')]}"/>
+                    </group>
+                </group>
+                <xpath expr='//div[@name="pricelist_item"]' position='inside'>
+                    <field name="item_ids" nolabel="1" context="{'default_base':'list_price', 'default_applied_on' :'1_product'}">
```
Comment (mdi-odoo) [inline addons/sale/views/sale_views.xml]: May be you can add this field from **product** module.

File: addons/event_sale/views/product_views.xml [hunk 16]
```diff
@@ -3,11 +3,13 @@
         <record id="event_sale_product_template_form" model="ir.ui.view">
              <field name="name">product.template.event.form.inherit</field>
              <field name="model">product.template</field>
-             <field name="inherit_id" ref="sale.product_template_form_view_invoice_policy" />
+             <field name="inherit_id" ref="product.product_template_form_view" />
              <field name="arch" type="xml">
-                <field name="invoice_policy" position="before">
-                    <field name="event_ok" />
-                </field>
+                <page name="sales" position="inside">
```
Comment (mdi-odoo) [inline addons/event_sale/views/product_views.xml]: Add after div name="pricelist_item" .

File: addons/purchase_requisition/views/purchase_requisition_views.xml [hunk 17]
```diff
@@ -256,11 +256,13 @@
     <record model="ir.ui.view" id="product_template_form_view_inherit">
         <field name="name">product.template.form.inherit</field>
         <field name="model">product.template</field>
-        <field name="inherit_id" ref="stock.view_template_property_form"/>
+        <field name="inherit_id" ref="purchase.view_product_supplier_inherit"/>
         <field name="arch" type="xml">
-            <field name="route_ids" position="after">
-                <field name="purchase_requisition" widget="radio"/>
-            </field>
+            <page name="purchase" position="inside">
```
Comment (mdi-odoo) [inline addons/purchase_requisition/views/purchase_requisition_views.xml]: Add after group name=**vendors** .
